src/components/wagonwheelComponent/index.js

```diff
@@ -52,10 +63,29 @@ class WagonWheelComponent extends Component {
     selectMatch (e){
         const match_id = parseInt(e.target.selectedOptions[0].value);
         const teamUrl = CONFIG.getTeams+"?match_id="+match_id;
+        const matchUrl = CONFIG.getAllDetails+"?id="+match_id;
         axios.get(teamUrl)
         .then((response)=> {
             this.setState({teamList :response.data})
-            this.setState({match_id : match_id})
+            this.setState({match_id : match_id});
+        })
+        .catch(function (error) {
+            console.log(error);
+        });
+
+        //populate match details
+
+        axios.get(matchUrl)
+        .then((response)=> {
+            if(this.props.onUpdateHeaderData)
+            {   
+                let data =  response.data[0];
+                this.props.onUpdateHeaderData({
```

Comment:
Check whether the function name is meaningful on the context. Confirm whether the on prefix is really needed.